index.ts

```diff
@@ -0,0 +1,65 @@
+"use strict";
+
+let onDisappearCallbacks: (() => void)[] = [];
```

Comment:
These could be WeakSets instead of arrays. That makes sure the callbacks do not leak memory.

Reply:
As discussed offline, WeakSets are not iterable so cannot be used here